docs/how-to/stream_ordered_allocator.rst

```diff
@@ -0,0 +1,423 @@
+
+    .. code-block:: cpp
+
+      #include <iostream>
+      #include <hip/hip_runtime.h>
+
+      // Kernel to perform some computation on allocated memory.
+      __global__ void myKernel(int* data, size_t numElements) {
+          int tid = threadIdx.x + blockIdx.x * blockDim.x;
+          if (tid < numElements) {
+              data[tid] = tid * 2;
+          }
+      }
+
+      int main() {
+          // Initialize HIP.
+          hipInit(0);
+
+          // Stream 0.
+          constexpr hipStream_t streamId = 0;
+
+          // Allocate memory with stream ordered semantics.
+          constexpr size_t numElements = 1024;
+          int* devData;
+          hipMallocAsync(&devData, numElements * sizeof(*devData), streamId);
+
+          // Launch the kernel to perform computation.
+          dim3 blockSize(256);
+          dim3 gridSize((numElements + blockSize.x - 1) / blockSize.x);
+          myKernel<<<gridSize, blockSize>>>(devData, numElements);
+
+          // Free memory with stream ordered semantics.
+          hipFreeAsync(devData, streamId);
+
+          // Synchronize to ensure completion.
+          hipDeviceSynchronize();
+
+          return 0;
+      }
+
+  .. tab-item:: Ordinary Allocation
+
+    .. code-block:: cpp
+
+      #include <iostream>
+      #include <hip/hip_runtime.h>
+
+      // Kernel to perform some computation on allocated memory.
+      __global__ void myKernel(int* data, size_t numElements) {
+          int tid = threadIdx.x + blockIdx.x * blockDim.x;
+          if (tid < numElements) {
+              data[tid] = tid * 2;
+          }
+      }
+
+      int main() {
+          // Initialize HIP.
+          hipInit(0);
+
+          // Allocate memory.
+          constexpr size_t numElements = 1024;
+          int* devData;
+          hipMalloc(&devData, numElements * sizeof(*devData));
+
+          // Launch the kernel to perform computation.
+          dim3 blockSize(256);
+          dim3 gridSize((numElements + blockSize.x - 1) / blockSize.x);
+          myKernel<<<gridSize, blockSize>>>(devData, numElements);
+
+          // Free memory.
+          hipFree(devData);
+
+          // Synchronize to ensure completion.
+          hipDeviceSynchronize();
+
+          return 0;
+      }
+
+For more details, visit the :ref:`stream ordered allocator reference <stream-ordered-allocator-reference>`.
+
+Memory pools
+============
+
+Memory pools provide a way to manage memory with stream-ordered behavior, ensuring proper synchronization and avoiding memory access errors. Division of a single memory system into separate pools allows querying the access path properties of each partition. Memory pools are used for host memory, device memory, and unified memory.
+
+Set pools
+---------
+
+The ``hipMallocAsync()`` function uses the current memory pool, while also providing the opportunity to create and use different pools with the ``hipMemPoolCreate()`` and ``hipMallocFromPoolAsync()`` functions respectively.
```

Comment:
My suggestion, to avoid word duplication in a sentence:

The ``hipMallocAsync()`` function uses the current memory pool and also provides the opportunity to create and access different pools using ``hipMemPoolCreate()`` and ``hipMallocFromPoolAsync()`` functions respectively.

Reply:
I can't spot the duplication you are talking about.

Reply:
"**use** different pools **using**"

Reply:
I think its ok. If you want, you can change it to create and utilize different pools...although the former one works well for me.